app/controllers/application_controller.rb

```diff
@@ -2,4 +2,16 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_filter :require_login
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+  private
+  def not_authenticated
+    redirect_to login_path, alert: 'Пожалуйста авторизуйтесь.'
+  end
+
+  def not_found
+    flash[:alert] = 'Вы обратились к несуществующей записи.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.